src/Confluent.Kafka/Producer.cs

```diff
@@ -293,124 +309,323 @@ public Producer(IEnumerable<KeyValuePair<string, object>> config, bool manualPol
         {
             if (!manualPoll)
             {
                 throw new InvalidOperationException("Poll method called when manualPoll not enabled.");
             }
-            this.kafkaHandle.Poll((IntPtr)millisecondsTimeout);
+            return this.kafkaHandle.Poll((IntPtr)millisecondsTimeout);
         }
 
-        public void Poll(TimeSpan timeout)
+        /// <summary>
+        ///     Poll for callback events. You will not typically need
+        ///     to call this method. Only call on producer instances 
+        ///     where background polling has been disabled.
+        /// </summary>
+        /// <param name="timeout">
+        ///     The maximum period of time to block if no callback events
+        ///     are waiting.
+        /// </param>
+        /// <returns>
+        ///     Returns the number of events served.
+        /// </returns>
+        public int Poll(TimeSpan timeout)
             => Poll(timeout.TotalMillisecondsAsInt());
 
-        public void Poll()
+        /// <summary>
+        ///     Poll for callback events. You will not typically need
+        ///     to call this method. Only call on producer
+        ///     instances where background polling has been disabled.
+        ///     Blocks until there is a callback event ready to be served.
+        /// </summary>
+        /// <returns>
+        ///     Returns the number of events served.
+        /// </returns>
+        public int Poll()
             => Poll(-1);
 
 
         /// <summary>
-        ///     Raised on critical errors, e.g. connection failures or all brokers down.
+        ///     Raised on critical errors, e.g. connection failures or all 
+        ///     brokers down.
         /// </summary>
+        /// <remarks>
+        ///     Called on the Producer poll thread.
+        /// </remarks>
         public event EventHandler<Error> OnError;
 
         /// <summary>
         ///     Raised on librdkafka stats events.
         /// </summary>
         /// <remarks>
-        ///     librdkafka statistics can be set using the statistics.interval.ms parameter.
+        ///     You can enable statistics and set the statistics interval
+        ///     using the statistics.interval.ms configuration parameter
+        ///     (disabled by default).
+        ///
+        ///     Called on the Producer poll thread.
         /// </remarks>
         public event EventHandler<string> OnStatistics;
 
         /// <summary>
-        ///     Raised when there is information that should be to be logged.
+        ///     Raised when there is information that should be logged.
         /// </summary>
         /// <remarks>
-        ///     Specify which log level with the log_level configuration property.
-        ///     can happen on any thread.
+        ///     You can specify the log level with the 'log_level' configuration
+        ///     property. You also probably want to specify one or more debug
```

Comment:
I'll add your comment that logging should not typically be enabled. However, i did want it and did find it useful at one point, and the need to specify a debug context tripped me up, so keen to leave that in.